renderer/reducers/backup.js

```diff
@@ -0,0 +1,180 @@
+import set from 'lodash.set'
+import { send } from 'redux-electron-ipc'
+import { grpcService } from 'workers'
+import { walletSelectors } from './wallet'
+import { infoSelectors } from './info'
+
+const SET_PROVIDER = 'SET_PROVIDER'
+
+const dbGet = async walletId => {
+  const wallet = await window.db.wallets.get(walletId)
+  return wallet && wallet.backup
+}
+
+/**
+ * Convenience wrapper that tries to update existing DB record and if fails insert new one
+ *
+ * @param {string} walletId
+ * @param {Object} update
+ */
+const dbUpdate = async (walletId, update) => {
+  return await window.db.wallets.update(walletId, { backup: update })
+}
+
+const dbTransaction = operation => {
+  return window.db.transaction('rw', window.db.wallets, operation)
+}
+
+/**
+ * IPC callback for backup service tokens update event
+ *
+ * @export
+ * @param {*} event
+ * @param {*} { provider, tokens, walletId }
+ */
+export function backupTokensUpdated(event, { provider, tokens, walletId }) {
+  return async () => {
+    await dbTransaction(async () => {
```

Comment:
Do this really need to be wrapped in a db transaction?

same question for all other instances of `dbTransaction` in this file

Reply:
Yes, because there `dbGet` and `dbSet` are async actions and race condition is possible (another request updates db state in between).